### ISSUE TITLE
More interactive in UI connector documentations

### ISSUE THREAD

**Issue by timroes** [airbytehq/airbyte#11434]
As part of our onboarding improvements we'd like to make sure the documentation we show for a user in the product will be more helpful and guiding them more directly about the steps they need to take to get the connector set up correctly. We'd therefore need to be able to link parts (paragraphs/steps) of the documentation to specific parts (fields) of the UI.

The desired UX would be that a user has automatically when selecting a connector a side-by-side view with the documentation. While they are reading through the documentation it might state e.g. "Copy this token and enter it as 'API key'". A click on "API key" should then focus the API key input of the connector form. Vice-versa when clicking into an input field of the form, we want to scroll and highlight the section in the documentation on the side mentioning which data needs to go into that input.

For that we'll need to reference fields from within the documentation. I believe this will require the following changes:

* Separate in UI documentation from our online documentation. I can imagine our wording could be way improved if we know it's in the product and we are able to link to the fields directly. @Amruta-Ranade Your thoughts around this would be highly welcome.
* Enable versioned documentation (for the in UI documentation). If there's a direct link between the documentation and UI, we must make sure the documentation we show actually matches the connector you're using. cc @sherifnada I know that we had some discussions around this already started.
* Agree on a syntax in the markdown that we want to use for linking it together with the form.

cc @misteryeo @nataliekwong

**Comment by Amruta-Ranade** [comment]
Can we set up some time to walk through a couple connector pages in the docs and the UI? I'd love to know the technical possibilities/limitation and have ideas and concerns about the project that might be easier to discuss in a meeting than asynchronously.

**Comment by nataliekwong** [comment]
@Amruta-Ranade and I chatted on Friday live. Some points I want to clarify: 
- Each form field should relate to a bookmark within the setup guide. Clicking on the form field will jump to the bookmark in the corresponding part of the docs on the right-hand side. We aren't necessarily removing content, as the user would need to go through some setup even prior to the form, that would still need to be done. Instead it's a clearer incremental navigation experience. 
- Users should still be able to navigate up and down through the complete setup guide, since they might need to reference previous information or want to skip ahead.

**Comment by timroes** [comment]
Summary from talking offline about this:

* We'd want to use the same markdown for documentation and in-product documentation for now.
* We'll add a feature allowing to use links in the format `[link text](#field:name)` to link to a specific field. In the UI we want to focus the field with the given name when the link is clicked. In our documentation generation we want to throw away those links and simply render the link text instead.
* We want to allow specific sections to be only rendered cloud only or OSS only in the product. We'll enable specifying specific regions by `<!-- ENV:cloud -->` `<!-- /ENV:cloud -->` and `<!-- ENV:oss -->` `<!-- /ENV:oss -->` to only be rendered in the product if it's cloud or OSS respectively. Since those would be HTML comments we don't need to change our documentation generation, they'll simply render out both sections for now. We can reconsider making this more interactive (e.g. using mdx components) later.

**Comment by timroes** [comment]
Blocked by #10644

**Comment by nataliekwong** [comment]
Should we close this now now that this is on connectors teams to implement? cc @flash1293

**Comment by girarda** [comment]
@nataliekwong is there another issue tracking the request?

**Comment by nataliekwong** [comment]
Issues for the connectors team? Not that I'm aware of.  I imagine it would be best implemented as individual issues (1 per connector or type of connector). I think it's okay to close this issue for now if the implementation piece is complete, and I would defer to the connector teams to determine the best way for their teams to manage adding this to the docs metadata. cc @katmarkham for API sources,  @prateekmukhedkar for DB sources, @evantahler for destinations.

**Comment by evantahler** [comment]
Are there any docs or examples about how to update the docs or specs?  I don't think we've socialized that this is a thing very well yet.

**Comment by flash1293** [comment]
We have versioned docs, the collapsable sections and "scroll to section" features now, I think we can close this issue and open new issues for the missing bits. I opened these two, feel free to add more as required:
* https://github.com/airbytehq/airbyte/issues/31436
* https://github.com/airbytehq/airbyte/issues/31435
